src/Microsoft.Extensions.Caching.Memory/MemoryCacheOptions.cs

```diff
@@ -2,22 +2,29 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using Microsoft.Extensions.Internal;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.Extensions.Caching.Memory
 {
-    public class MemoryCacheOptions : IOptions<MemoryCacheOptions>
+    public class MemoryCacheOptions
     {
+        public MemoryCacheOptions(IEnumerable<IConfigureOptions<MemoryCacheOptions>> configureOptions = null)
```

Comment:
I thought you were going to try having a base class with the boilerplate constructor. Did you try? I would ok either way, but it would be good to see how it looks like.